TerraformAsset causes apply to fail when referencing typescript projects

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### cdktf & Language Versions

<!--- Please make note of the version you are using for the `hashicorp/terraform-cdk` and language bindings you are using.
If you are not running the latest version, please upgrade because your issue may have already been fixed. --->

I'm using CDKTF 0.4 and Typescript 4.3.2

### Affected Resource(s)

<!--- Please list the affected resources. --->

- TerraformAsset

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
--->


```log
> cdktf apply --auto-approve

cdktf.out/stacks/aws/assets/backend-project/src/config.ts:1:8 - error TS1192: Module '"/Users/danielschmidt/work/cdktf-integration-docker-example/infrastructure/aws/cdktf.out/stacks/aws/assets/backend-project/node_modules/@types/pg/index"' has no default export.

1 import pg from "pg";
```

### Expected Behavior

<!--- What should have happened? --->

`cdktf apply` should not compile my assets

### Actual Behavior

<!--- What actually happened? --->

It tries to and fails

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

- Create a new repo with the typescript template
- create a new typescript project
- reference the typescrupt project with TerraformAsset
- run `cdktf apply`

### Workaround

Change your tsconfig to exclude the cdktf.out folder:

```json
{
  "exclude": ["node_modules", "cdktf.out"]
}
```

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/packages/cdktf-cli/templates/typescript/tsconfig.json
+++ b/packages/cdktf-cli/templates/typescript/tsconfig.json
@@ -28,6 +28,7 @@
     "**/*.ts"
   ],
   "exclude": [
-    "node_modules"
+    "node_modules",
+    "cdktf.out"
   ]
 }

